algorithms_task_4/student/include/impl.h

```diff
@@ -12,4 +13,20 @@
  * @return new slided range position
  */
 template <typename RandomAccessIterator>
-std::pair<RandomAccessIterator, RandomAccessIterator> slide(RandomAccessIterator begin, RandomAccessIterator last, RandomAccessIterator pos);
+std::pair<RandomAccessIterator, RandomAccessIterator> slide(RandomAccessIterator begin, RandomAccessIterator last, RandomAccessIterator pos)
+{
+    if (pos == begin || pos == last) {
+        return { begin, last };
+    }
+
+    bool isRightShift =  pos > last;
+    if (isRightShift) {
+        auto newBegin = std::rotate(begin, last, pos);
+        return { newBegin, pos };
+    } else {
+        using rev_it = std::reverse_iterator<RandomAccessIterator>;
+
+        auto newEnd = std::rotate(rev_it(last), rev_it(begin), rev_it(pos));
+        return { pos, newEnd.base() };
```

Comment:
Please avoid conversion "reverse iterator to normal".
This is wrong. You could add test, where newEnd will be the rbegin/rend, and you will see the problem.
Try to accomplish this task with plain iterators